File: scientific calculator.c
#include <stdio.h>
#include <math.h>

void add(double x, double y) {
    printf("%.2f + %.2f = %.2f\n", x, y, x + y);
}

void subtract(double x, double y) {
    printf("%.2f - %.2f = %.2f\n", x, y, x - y);
}

void multiply(double x, double y) {
    printf("%.2f * %.2f = %.2f\n", x, y, x * y);
}

void divide(double x, double y) {
    if (y == 0) {
        printf("Error! Division by zero.\n");
    } else {
        printf("%.2f / %.2f = %.2f\n", x, y, x / y);
    }
}

void power(double base, double exp) {
    printf("%.2f ^ %.2f = %.2f\n", base, exp, pow(base, exp));
}

void square_root(double x) {
    if (x < 0) {
        printf("Error! Square root of negative number.\n");
    } else {
        printf("sqrt(%.2f) = %.2f\n", x, sqrt(x));
    }
}

void sine(double angle) {
    printf("sin(%.2f degrees) = %.2f\n", angle, sin(angle * M_PI / 180));
}

void cosine(double angle) {
    printf("cos(%.2f degrees) = %.2f\n", angle, cos(angle * M_PI / 180));
}

void tangent(double angle) {
    printf("tan(%.2f degrees) = %.2f\n", angle, tan(angle * M_PI / 180));
}

int main() {
    int choice;
    double num1, num2;

    do {
        printf("\nSelect operation:\n");
        printf("1. Add\n");
        printf("2. Subtract\n");
        printf("3. Multiply\n");
        printf("4. Divide\n");
        printf("5. Power\n");
        printf("6. Square Root\n");
        printf("7. Sine\n");
        printf("8. Cosine\n");
        printf("9. Tangent\n");
        printf("10. Exit\n");

        printf("Enter choice (1-10): ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                printf("Enter first number: ");
                scanf("%lf", &num1);
                printf("Enter second number: ");
                scanf("%lf", &num2);
                add(num1, num2);
                break;
            case 2:
                printf("Enter first number: ");
                scanf("%lf", &num1);
                printf("Enter second number: ");
                scanf("%lf", &num2);
                subtract(num1, num2);
                break;
            case 3:
                printf("Enter first number: ");
                scanf("%lf", &num1);
                printf("Enter second number: ");
                scanf("%lf", &num2);
                multiply(num1, num2);
                break;
            case 4:
                printf("Enter first number: ");
                scanf("%lf", &num1);
                printf("Enter second number: ");
                scanf("%lf", &num2);
                divide(num1, num2);
                break;
            case 5:
                printf("Enter base: ");
                scanf("%lf", &num1);
                printf("Enter exponent: ");
                scanf("%lf", &num2);
                power(num1, num2);
                break;
            case 6:
                printf("Enter number: ");
                scanf("%lf", &num1);
                square_root(num1);
                break;
            case 7:
            case 8:
            case 9:
                printf("Enter angle in degrees: ");
                scanf("%lf", &num1);
                if (choice == 7)
                    sine(num1);
                else if (choice == 8)
                    cosine(num1);
                else
                    tangent(num1);
                break;
            case 10:
                printf("Exiting the calculator.\n");
                break;
            default:
                printf("Invalid input. Please enter a number between 1 and 10.\n");
        }
    } while (choice != 10);

    return 0;
}

File: scientific calculator.py
import math

def exponentiate(x, y):
    return x ** y

def square_root(x):
    return math.sqrt(x)

def sine(x):
    return math.sin(math.radians(x))

def cosine(x):
    return math.cos(math.radians(x))

def tangent(x):
    return math.tan(math.radians(x))

def logarithm_base_10(x):
    if x > 0:
        return math.log10(x)
    else:
        return "Error! Logarithm of non-positive number."

def natural_logarithm(x):
    if x > 0:
        return math.log(x)
    else:
        return "Error! Logarithm of non-positive number."

def main():
    print("Scientific Calculator (Advanced Functions Only)")
    print("Select operation:")
    print("1. Exponentiate")
    print("2. Square Root")
    print("3. Sine")
    print("4. Cosine")
    print("5. Tangent")
    print("6. Logarithm Base 10")
    print("7. Natural Logarithm")

    while True:
        choice = input("Enter choice (1-7) or 'q' to quit: ")

        if choice == 'q':
            break

        if choice in ['1']:
            num1 = float(input("Enter base: "))
            num2 = float(input("Enter exponent: "))
            print(f"{num1} ^ {num2} = {exponentiate(num1, num2)}")

        elif choice in ['2', '3', '4', '5', '6', '7']:
            num = float(input("Enter the number: "))

            if choice == '2':
                print(f"√{num} = {square_root(num)}")
            elif choice == '3':
                print(f"sin({num}) = {sine(num)}")
            elif choice == '4':
                print(f"cos({num}) = {cosine(num)}")
            elif choice == '5':
                print(f"tan({num}) = {tangent(num)}")
            elif choice == '6':
                print(f"log10({num}) = {logarithm_base_10(num)}")
            elif choice == '7':
                print(f"ln({num}) = {natural_logarithm(num)}")

        else:
            print("Invalid input!")

if __name__ == "__main__":
    main()
